[cli] klio message publish blocks on pubsub client future

DanSimon opened the pull request:
When using `google-cloud-pubsub` `2.5.0`, `klio message publish` will say that the message is successfully published, but the message is not actually published to the topic.  This appears to have been some underlying threading change causing a race condition where `klio-cli` doesn't wait on the returned `Future` and exits before the message is actually published.

Simply blocking on the future fixes the issue.  I tested with pubsub `2.0.0`, `2.3.0` and `2.5.0`.

This is just a quick fix and there is room for further improvement.  If multiple messages are being published, this will end up blocking for each message, which is a bit inefficient since the pubsub client does have internal batching capabilities, but for small numbers of messages it shouldn't be big deal.

## Checklist for PR author(s)
<!-- If an item doesn't apply to your pull request, **check it anyway** to make it apparent that there's nothing left to do. -->
- [ ] Format the pull request title like `[cli] Fixes bugs in 'klio job fake-cmd'`.
- [ ] Changes are covered by unit tests (no major decrease in code coverage %) and/or integration tests.
- [ ] Document any relevant additions/changes in the appropriate spot in `docs/src`.
- [ ] For any change that affects users, update the package's changelog in ``docs/src/reference/<package>/changelog.rst``.


<!---
for more information on how to submit valuable contributions,
see https://opensource.guide/how-to-contribute/#how-to-submit-a-contribution
-->

shireen-bean commented on cli/src/klio_cli/commands/message/publish.py:
```diff
@@ -89,7 +89,9 @@ def _publish_messages(
-            publish(data=message)
+            future = publish(data=message)
+            # block until message is published or exception is raised
+            future.result()
```
Just double checking: I know this was related to a library update but is `future.result()` backwards compatible?

DanSimon replied:
Yep, since we already require `google-cloud-pubsub>=2.0.0` and I tested successfully with `2.0.0`

shireen-bean reviewed (dismissed):
Mind adding that this was fixed in the changelog please?
So we can keep track of what's getting released in the fixes!
ty ty!!

DanSimon commented:
@shireen-bean wow I didn't even follow my own advice!  Added to the changelog, also added a commit to fix a docs link that apparently was just changed.

shireen-bean reviewed (approved):
Idk if this needed another ✅ before merging but thanks for adding the changelogs haha
